tests/scripts/validate_resources.py

```diff
@@ -0,0 +1,150 @@
+from __future__ import annotations
+import ast
+import json
+import os
+from typing import Any, Dict, Generator, List, Optional, Tuple
+
+
+from ocp_resources.resource import Resource  # noqa
+
+
+def _get_api_value_by_type(api_value: str, api_type: str) -> Optional[str]:
+    try:
+        if api_value:
+            if api_type == "api_version":
+                return eval(f"Resource.ApiVersion.{api_value}")
+
+            elif api_type == "api_group":
+                return eval(f"Resource.ApiGroup.{api_value}")
+
+    except AttributeError:
+        return None
+
+
+def _get_api_group_and_version(bodies: List[Any]) -> Tuple[str, str]:
+    for targets in bodies:
+        targets: ast.AnnAssign | ast.Attribute
+        if isinstance(targets, ast.AnnAssign):
+            api_type = targets.target.id
+        else:
+            api_type = targets.targets[0].id
+
+        return api_type, getattr(targets.value, "attr", getattr(targets.value, "value", None))
+
+    return "", ""
+
+
+def validate_resource(
+    cls: ast.ClassDef,
+    resource_list: List[Dict[str, Any]],
+    api_value: str,
+    api_type: str,
+) -> List[str]:
+    errors: List[str] = []
+    resource_str: str = "Resource"
+    namespaced_resource_str: str = "NamespacedResource"
+    _base_class_error: str = f"Resource {cls.name} must have {resource_str} or {namespaced_resource_str} base class"
+    _api_type_group: bool = api_type == "api_group"
+    _api_type_version: bool = api_type == "api_version"
+
+    bases = getattr(cls, "bases", [])
+    if not bases:
+        return [_base_class_error]
+
+    base_class_from = [_base for _base in bases if _base.id in (resource_str, namespaced_resource_str)]
+    if not base_class_from:
+        return [f"{_base_class_error}, got {[_base.id for _base in bases]}"]
+
+    if len(base_class_from) > 1:
+        return [f"Resource {cls.name} must have only one of {resource_str} or {namespaced_resource_str} base class"]
+
+    if not api_type:
+        return [f"Resource {cls.name} must have api_group or api_version"]
```

Comment:
we do not want to check base class as well?

Reply:
What do you mean?

Reply:
check if base class is the right one is done after this part